src/foam/nanos/auth/User.js

```diff
@@ -227,7 +232,26 @@ foam.CLASS({
       includeInDigest: true,
       containsPII: true,
       trim: true,
-      tableWidth: 160
+      tableWidth: 160,
+      validateObj: function(firstName) {
+        if ( ! firstName.trim() ) return;
+
+        var invalidMatcher = new RegExp(this.INVALID_MATCHER, 'ug');
+        var invalidMatch = firstName.match(invalidMatcher)
+        if ( invalidMatch ) {
+          var invalidChars = [...new Set(invalidMatch.join(''))].join(', ')
+          return this.INVALID_FIRST_NAME + invalidChars;
+        }
+      },
+      javaValidateObj: `
+        foam.nanos.app.AppConfig appConfig = (foam.nanos.app.AppConfig) x.get("appConfig");
+        String name = (String) obj.getProperty("firstName");
+        if ( name.length() == 0 || appConfig.getMode() == foam.nanos.app.Mode.TEST ) return;
+
+        java.util.regex.Pattern nameMatcher = java.util.regex.Pattern.compile((String) obj.getProperty("NAME_MATCHER"));
```

Comment:
Compiling patterns is very expensive, which is why once you've compiled one you would like to reuse it. It would be better if this were a Pattern constant than a string message.